pkg/container/containerd.go

```diff
@@ -166,3 +172,345 @@ func (d *DisableContainerd) Execute(runtime connector.Runtime) error {
 	}
 	return nil
 }
+
+type MigrateSelfNodeTocontainerd struct {
+	common.KubeAction
+}
+
+func (d *MigrateSelfNodeTocontainerd) Execute(runtime connector.Runtime) error {
+	nodeName, _ := runtime.GetRunner().SudoCmd("hostname", true)
+	if _, err := runtime.GetRunner().SudoCmd(fmt.Sprintf("/usr/local/bin/kubectl cordon %s ", nodeName), true); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("cordon the node: %s failed", nodeName))
+	}
+	if _, err := runtime.GetRunner().SudoCmd(fmt.Sprintf(
+		"/usr/local/bin/kubectl drain %s --delete-emptydir-data --ignore-daemonsets --timeout=2m --force", nodeName),
+		true); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("drain the node: %s failed", nodeName))
+	}
+	// if _, err := runtime.GetRunner().SudoCmd("systemctl stop kubelet", false); err != nil {
+	// 	return errors.Wrap(errors.WithStack(err), fmt.Sprintf("stop kubelet failed: %s", nodeName))
+	// }
+	if err := UninstallContainerTasks(runtime, d.KubeAction); err != nil {
+		return errors.Wrap(errors.WithStack(err), fmt.Sprintf("Uninstall ContainerTasks failed: %s", nodeName))
+	}
+
+	if err := MigrateToContainerdTasks(runtime, d.KubeAction); err != nil {
+		return errors.Wrap(errors.WithStack(err), fmt.Sprintf("Migrate To Containerd failed: %s", nodeName))
+	}
+	if _, err := runtime.GetRunner().SudoCmd(fmt.Sprintf(
+		"sed -i 's#--network-plugin=cni --pod#--network-plugin=cni --container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock --pod#' /var/lib/kubelet/kubeadm-flags.env"),
+		true); err != nil {
+		return errors.Wrap(err, "Change KubeletTo Containerd failed")
+	}
+	if _, err := runtime.GetRunner().SudoCmd(fmt.Sprintf("systemctl daemon-reload && systemctl restart containerd"), true); err != nil {
+		return errors.Wrap(err, "restart containerd")
+	}
+
+	f := true
+	for f {
+		if _, err := runtime.GetRunner().SudoCmd(fmt.Sprintf("/usr/local/bin/kubectl uncordon %s", nodeName), true); err == nil {
+			break
+		}
+
+	}
+	if _, err := runtime.GetRunner().SudoCmd(fmt.Sprintf("/usr/local/bin/kubectl uncordon %s", nodeName), true); err != nil {
+		return errors.Wrap(err, "restart kubelet and unDrain Node  failed")
+	}
+
+	return nil
+}
+
+func UninstallContainerTasks(runtime connector.Runtime, kubeAction common.KubeAction) error {
+	host := runtime.RemoteHost()
+	switch kubeAction.KubeConf.Cluster.Kubernetes.ContainerManager {
```

Comment:
It is a little strange. It means need the user to set up this value in the kk config first, and then specify the target cri in the command flag `--type containerd`. I don't know if it is good user-interaction logic.